go/src/pluginmgr/pluginmgr_common/pluginmgr_helper.go

```diff
@@ -0,0 +1,491 @@
+				retMsg, PLUGIN_INIT_PERIODIC_TIMEOUT_DEFAULT, PLUGIN_INIT_PERIODIC_FALLBACK_TIMEOUT_DEFAULT, "")
+		}
+	}()
+
+	// 1.Check if plugin is already loaded
+	if _, ok := plmgr.getPlugin(pluginName); ok {
+		retMsg = fmt.Sprintf("plugin with name %s and version %s is already loaded", pluginName, pluginVersion)
+		return lomcommon.LogError(retMsg)
+	}
+
+	// 2.Get plugin specific details from actions config file and add any additional info(future) to pass to plugin's init() call
+	actionCfg, err := lomcommon.GetConfigMgr().GetActionConfig(pluginName)
+	if err != nil {
+		retMsg = fmt.Sprintf("plugin %s not found in actions config file", pluginName)
+		return lomcommon.LogError(retMsg)
+	}
+
+	// 3.Check if plugin disabled flag is set or not in the actions config file.
+	if actionCfg.Disable {
+		lomcommon.LogWarning("Plugin %s is disabled", pluginName)
+		return nil
+	}
+
+	// 4.Create new plugin instance
+	pluginID := plugins_common.PluginId{Name: pluginName, Version: pluginVersion}
+	plugin, pluginmetadata, err := CreatePluginInstance(pluginID, actionCfg) // returns Plugin interface pointing to new plugin struct
+	if err != nil {
+		retMsg = fmt.Sprintf("Error creating plugin instance for %s %s: %s", pluginName, pluginVersion, err)
+		return lomcommon.LogError(retMsg)
+	}
+
+	// 5.Check if plugin name and version from proc_conf.json file matches the values in static plugin. If not log periodic log
+	if id := plugin.GetPluginID(); id.Name != pluginName || id.Version != pluginVersion {
+		retMsg = fmt.Sprintf("Plugin ID does not match provided arguments: got %s %s, expected %s %s", id.Name, id.Version, pluginName, pluginVersion)
+		return lomcommon.LogError(retMsg)
+	}
+
+	// 6. call plugin's init() call synchronously
+	err = plugin.Init(actionCfg)
+	if err != nil {
+		//delete(plmgr.plugins, pluginName) // TODO: Goutham : verify Is this needed instead?
+		//delete(plmgr.pluginMetadata, pluginName) // TODO: Goutham : verify Is this needed instead.
+		retMsg = fmt.Sprintf("plugin %s init failed: %v", pluginName, err)
+		return lomcommon.LogError(retMsg)
+	}
+
+	// 7. call plugin's registerAction() call synchronously
+	err = plmgr.registerActionWithEngine(pluginName)
+	if err != nil {
+		//delete(plmgr.plugins, pluginName) // TODO: Goutham : verify Is this needed instead?
+		//delete(plmgr.pluginMetadata, pluginName) // TODO: Goutham : verify Is this needed instead.
+		retMsg = fmt.Sprintf("plugin %s registerAction failed: %v", pluginName, err)
+		return lomcommon.LogError(retMsg)
+	}
+
+	// 8.Add plugin to plugin manager's map
+	plmgr.plugins[pluginName] = plugin
+	plmgr.pluginMetadata[pluginName] = pluginmetadata
+
+	return nil
+}
+
+func (plmgr *PluginManager) loadPlugin(pluginName string, pluginVersion string) error {
+	// Create a channel to receive the result of AddPlugin()
+	resultChan := make(chan error)
+
+	lomcommon.GetGoroutineTracker().Start("plg_mgr_LoadPlugin"+pluginName, func() {
```

Comment:
can make it const.

Reply:
its just unique name given to goroutine ... simple convention is functionname+XXX given at the call to goroutine.tracker . And it will not be used anywhere.